### ISSUE TITLE
lstlisting language formatting considered annoying

### ISSUE THREAD

**Issue by mbtaylor** [ivoa-std/ivoatexDoc#8]
I generally find the language-specific formatting done by `lstlisting` more annoying than beneficial.  For instance in sec 4.2 of the document, the following stanza:
```
\begin{lstlisting}[language=sh]
git checkout main
git pull origin main
git checkout -b branch-name-for-change
\end{lstlisting}
```
formats it all monospaced apart from the "for" which is in bold face.  In section 2, this one:
```
\begin{lstlisting}[language=sh,basicstyle=\footnotesize]
git clone --recurse-submodules https://github.com/msdemlei/ivoatexDoc
cd ivoatexDoc
make biblio   # update the bibliography
make forcetex # make a PDF ignoring timestamps
make ivoatexDoc.html  # make an html document
make package # make a zipfile for IVOA submission
\end{lstlisting}
```
for some reason typesets "cd" in bold, and the rest of it in monospaced although it does the comments (after the #) in italic, which is arguably an improvement.

I find this (not very) cleverness distracting and would generally just prefer the whole thing monospaced including the comments.  Arguably the benefits are greater for language settings other than `sh`, though e.g. it doesn't seem to be doing anything particularly smart for the archdiag example in sec 2.3.3 marked `language=XML`.

Is it just me?  It's partly a matter of taste, and I can cope either way, but given we're recommending particular usage of the `lstlisting` package in ivoatexDoc I thought it was worth raising the question.

**Comment by msdemlei** [comment]
It would seem that for some reason a comment on this I sent on March 21
never made it here.  So... let me try again:

> I generally find the language-specific formatting done by
> `lstlisting` more annoying than beneficial.  For instance in sec

Well, I can't say I'm a big fan of syntax highlighting myself, so
I'll certainly not oppose some mild discouragement of using language
in the lstlistings options.

The main reason I'm urging people to use lstlistings is that people
are usually unhappy with verbatim ("font too large"), and changing
the built-in verbatim *is* a bit tricky.  Then, people start to use
all kinds alternative verbatim environments, and for those, making
HTML is particularly annoying given tth's inability to follow catcode
tricks.  So, I thought let's adopt one verbatim environment and
provide reasonalbe HTML support of that one environment.  lstlistings
seemed the one that's best thought out.

> 4.2 of the document, the following stanza:
> ```
> \begin{lstlisting}[language=sh]
> git checkout main
> git pull origin main
> git checkout -b branch-name-for-change
> \end{lstlisting}
> ```
> formats it all monospaced apart from the "for" which is in bold
> face.  In section 2, this one:

Yeah, that's lame.  It could have figured out that's not a shell
builtin, really.

> ```
> \begin{lstlisting}[language=sh,basicstyle=\footnotesize]
> git clone --recurse-submodules https://github.com/msdemlei/ivoatexDoc
> cd ivoatexDoc
> make biblio   # update the bibliography
> make forcetex # make a PDF ignoring timestamps
> make ivoatexDoc.html  # make an html document
> make package # make a zipfile for IVOA submission
> \end{lstlisting}
> ```
> for some reason typesets "cd" in bold, and the rest of it in
> monospaced although it does the comments (after the #) in italic,
> which is arguably an improvement.

Well, the cd is a shell builtin, and the rest isn't, so that at least
has some logic.

> Is it just me?  It's partly a matter of taste, and I can cope
> either way, but given we're recommending particular usage of the
> `lstlisting` package in ivoatexDoc I thought it was worth raising
> the question.


As I'm arguing above, throwing lstlistings out entirely is, I think,
not an option.

But I'd certainly merge a PR taking out most of the language=X
attributes from ivoatexDoc, and also one that discourages people from
overdoing syntax highlighting.  Except for the "for" above (which is
bad, and I should figure out how to fix this), however, I can't find
syntax highlighting terribly distracting, so I'll not spend time on
moving in that direction myself.

**Comment by Zarquan** [comment]
If you leave the language blank, `lstlisting` doesn't highlight anything.
```
\begin{lstlisting}[]
   This will be in monotype font with no highlighting.
\end{lstlisting}
```

**Comment by msdemlei** [comment]
Fixed in PR #14 by Mark himself.  Thanks!

**Comment by msdemlei** [comment]
On Mon, Mar 21, 2022 at 05:37:59AM -0700, Mark Taylor wrote:
> I generally find the language-specific formatting done by
> `lstlisting` more annoying than beneficial.  For instance in sec

Well, I can't say I'm a big fan of syntax highlighting myself, so
I'll certainly not oppose some mild discouragement of using language
in the lstlistings options.

The main reason I'm urging people to use lstlistings is that people
are usually unhappy with verbatim ("font too large"), and changing
the built-in verbatim *is* a bit tricky.  Then, people start to use
all kinds alternative verbatim environments, and for those, making
HTML is particularly annoying given tth's inability to follow catcode
tricks.  So, I thought let's adopt one verbatim environment and
provide reasonalbe HTML support of that one environment.  lstlistings
seemed the one that's best thought out.

> 4.2 of the document, the following stanza:
> ```
> \begin{lstlisting}[language=sh]
> git checkout main
> git pull origin main
> git checkout -b branch-name-for-change
> \end{lstlisting}
> ```
> formats it all monospaced apart from the "for" which is in bold
> face.  In section 2, this one:

Yeah, that's lame.  It could have figured out that's not a shell
builtin, really.

> ```
> \begin{lstlisting}[language=sh,basicstyle=\footnotesize]
> git clone --recurse-submodules https://github.com/msdemlei/ivoatexDoc
> cd ivoatexDoc
> make biblio   # update the bibliography
> make forcetex # make a PDF ignoring timestamps
> make ivoatexDoc.html  # make an html document
> make package # make a zipfile for IVOA submission
> \end{lstlisting}
> ```
> for some reason typesets "cd" in bold, and the rest of it in
> monospaced although it does the comments (after the #) in italic,
> which is arguably an improvement.

Well, the cd is a shell builtin, and the rest isn't, so that at least
has some logic.

> Is it just me?  It's partly a matter of taste, and I can cope
> either way, but given we're recommending particular usage of the
> `lstlisting` package in ivoatexDoc I thought it was worth raising
> the question.

As I'm arguing above, throwing lstlistings out entirely is, I think,
not an option.

But I'd certainly merge a PR taking out most of the language=X
attributes from ivoatexDoc, and also one that discourages people from
overdoing syntax highlighting.  Except for the "for" above (which is
bad, and I should figure out how to fix this), however, I can't find
syntax highlighting terribly distracting, so I'll not spend time on
moving in that direction myself.
